server/sonar-server/src/main/java/org/sonar/server/webhook/ws/SearchAction.java

```diff
@@ -0,0 +1,136 @@
+ */
+package org.sonar.server.webhook.ws;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Resources;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.server.setting.ws.Setting;
+import org.sonar.server.setting.ws.SettingsFinder;
+import org.sonar.server.user.UserSession;
+import org.sonarqube.ws.Webhooks.SearchWsResponse.Builder;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+import static org.sonar.api.web.UserRole.ADMIN;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.ORGANIZATION_KEY_PARAM;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.PROJECT_KEY_PARAM;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.SEARCH_ACTION;
+import static org.sonar.server.ws.KeyExamples.KEY_ORG_EXAMPLE_001;
+import static org.sonar.server.ws.KeyExamples.KEY_PROJECT_EXAMPLE_001;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.Webhooks.SearchWsResponse.newBuilder;
+
+public class SearchAction implements WebhooksWsAction {
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final SettingsFinder settingsFinder;
+
+  public SearchAction(DbClient dbClient, UserSession userSession, SettingsFinder settingsFinder) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.settingsFinder = settingsFinder;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+
+    WebService.NewAction action = controller.createAction(SEARCH_ACTION)
+      .setDescription("Search for webhooks associated to an organization or a project.<br/>")
+      .setSince("7.1")
+      .setResponseExample(Resources.getResource(this.getClass(), "example-webhooks-search.json"))
+      .setHandler(this);
+
+    action.createParam(ORGANIZATION_KEY_PARAM)
+      .setDescription("Organization key. If no organization is provided, the default organization is used.")
+      .setInternal(true)
+      .setRequired(false)
+      .setExampleValue(KEY_ORG_EXAMPLE_001);
+
+    action.createParam(PROJECT_KEY_PARAM)
+      .setDescription("Project key")
+      .setRequired(false)
+      .setExampleValue(KEY_PROJECT_EXAMPLE_001);
+
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+
+    String project = request.param(PROJECT_KEY_PARAM);
+
+    userSession.checkLoggedIn();
+
+    List<SearchElement> elements = doHandle(project);
+
+    writeResponse(request, response, elements);
+
+  }
+
+  private List<SearchElement> doHandle(String project) {
```

Comment:
missing "@Nullable String project". Note that the parameter could be renamed "projectKey" to be more precise.

Reply:
done